### ISSUE TITLE
gnome-screenshot: Uses Adwaita when prefer-dark in ON

### ISSUE THREAD

**Issue by Qronikarz** [linuxmint/mint21.2-beta#16]
While I was making screenshot for previous issue I noticed that screenshot program uses wrong accent color so I investigated it.
Here are the results:

I switched to red accent color so it's visible better. I use Cinnamon, dark mode Mint-Y.
When you have "Prefer Dark mode" in theme settings it looks like this:
![Zrzut ekranu z 2023-06-22 00-26-49](https://github.com/linuxmint/mint21.2-beta/assets/84787215/7eb00622-3c64-44e3-9ab3-149568b4184e)
You can notice the save screenshot button and the file name get a dark blue color accent even though red should be used.

When you switch to "prefer light mode" in theme settings it switches to something like this:
![Zrzut ekranu z 2023-06-22 00-27-05](https://github.com/linuxmint/mint21.2-beta/assets/84787215/74a0f1aa-901f-46f4-9db7-cd64409e700b)
Now the file name gets a correct accent color, but the save button is in the color of old green before Mint 21.1

**Comment by clefebvre** [comment]
It looks like it uses libAdwaita... is that the screenshot program we ship with? i.e. gnome-screenshot?

**Comment by clefebvre** [comment]
Confirm, it must be using libhandy or something. When Prefer Dark is ON, these libs stupidly switch to Adwaita-Dark. This is super annoying.

**Comment by Qronikarz** [comment]
Bad news. When extracting my archives I noticed that this program also looks out of place.
I switched to English language and tested all included by default programs and here's a list of those having the same problem:
* Archive Manager
* Calculator
* Calendar
* Celluloid (using old green "Save" button)
* Disks
* Disk Usage Analyzer
* Document Scanner
* Logs
* Passwords and Keys
* Printers
* Screenshot
* System Monitor
* Welcome Screen (using old green for "Launch" buttons)

The names are from Cinnamon main menu.

**Comment by clefebvre** [comment]
for some of them I think it might be because our dark themes are missing a gtk-dark.css.

**Comment by haggen88** [comment]
I confirm, when changing to dark theme some apps change to Adawaita-dark.

--

P.S. @clefebvre  I have a question, is it normal that having 1 additional screen connected, when taking screenshot, gnome-screnshot combines both screens into 1 [image](https://i.ibb.co/gzFByZp/Screenshot-from-2023-06-22-17-57-48.png)?

**Comment by anaximeno** [comment]
> P.S. @clefebvre I have a question, is it normal that having 1 additional screen connected, when taking screenshot, gnome-screnshot combines both screens into 1 [image](https://i.ibb.co/gzFByZp/Screenshot-from-2023-06-22-17-57-48.png)?

It worked this way before.

**Comment by kacperpaczos** [comment]
> Confirm, it must be using libhandy or something. When "Prefer Dark" is ON, these libs stupidly switch to Adwaita-Dark. This is super annoying.

I suggest creating a fork of gnome-screenshot to fix this and other issues, because gnome-screenshot is no longer supported.
https://gitlab.gnome.org/GNOME/gnome-screenshot/-/merge_requests/62

I think it's a great utility for LM users, and we can fix this and create the possibility to develop it according to our vision of the system.

@clefebvre

**Comment by axel358** [comment]
I can only reproduce this if both a dark theme and Prefer dark mode are set. If u have a light theme + Prefer dark libhandy apps go that theme's dark version, if u have a Dark theme + Let applications decide libhandy apps use that dark theme. Sw

**Comment by axel358** [comment]
Update: If u use a Mint-L theme the bug occurs even with light themes

**Comment by axel358** [comment]
The Mint-L lacks gtk-dark.css on any variant. Creating or linking a gtk-dark.css fixes the issue, same for the Mint-Y themes, adding a gtk-dark.css inside the dark theme variants fixes the issue

**Comment by Uldreth** [comment]
> Confirm, it must be using libhandy or something. When Prefer Dark is ON, these libs stupidly switch to Adwaita-Dark. This is super annoying.

@clefebvre  I think the problem is the naming of the theme. The current standard GTK expects is that if a theme with name "ThemeName" has both a light and a dark variant, then the name of the dark variant should be ThemeName-dark. Including the "-dark" suffix being lowercase. See also my comment  [here](https://github.com/linuxmint/mint-themes/issues/420#event-9364086520) .

Now I am not a dev and certainly not a GTK dev, but from what I understand that if the "Prefer Dark" setting is set, some GTK3 apps (I guess the ones that use libhandy) will look for ThemeName and if they find a ThemeName-dark, they will use the assets of that theme, otherwise they will revert to the default dark theme, in this case Adwaita.

Indeed I have just checked in a VM running Mint 21.2 beta that if you copy eg. /usr/share/themes/Mint-Y-Dark-Aqua and rename it to /usr/share/themes/Mint-Y-Aqua-dark (I have also changed the names in the index.theme, not sure if it is needed), then use this renamed theme with the Prefer Dark setting enabled, then standard GTK3 and libhandy (Disks in the image I am linking) applications will both use the correct theme and in addition to that, libadwaita apps (Flatseal in the image) also correctly get the dark theme (although in the latter case obviously it is the standard libadwaita theme).
![xjkzWDo](https://github.com/linuxmint/mint21.2-beta/assets/29836149/32e9e3b4-1a89-471c-9cb5-7804cbdb7fd2)

**Comment by JosephMcc** [comment]
> @clefebvre I think the problem is the naming of the theme. The current standard GTK expects is that if a theme with name "ThemeName" has both a light and a dark variant, then the name of the dark variant should be ThemeName-dark. Including the "-dark" prefix being lowercase. See also my comment https://github.com/linuxmint/mint-themes/issues/420#event-9364086520 .

I can confirm this. I use a custom theme and it works just fine following this convention. You'll also notice that this is the convention used by Adwaita now.

**Comment by mdevansh06** [comment]
This happens basically in all gnome apps. I installed nautilus and evince and happens there too, not only in victoria but in vera also, and in other desktops like budgie also. So this is not a mint or cinnamon problem. It is just a weird naming convention.


I have also tried editing the themes or the themes which use the name convention and they work fine.


A possible solution may be what Ubuntu budgie did i.e., use mate apps in place of corresponding gnome ones.
